backend/services/theme_service.py

```
@@ -77,97 +77,343 @@ def _create_smart_person_background_mask(self, image_bytes: bytes, image_width:
+            for corner in corners:
+                if corner.size > 0:  # Ensure corner is not empty
+                    background_colors.append(np.mean(corner, axis=(0, 1)))
+            
+            if not background_colors:
+                raise ValueError("Could not extract background colors from image corners")
+                
+            background_mean = np.mean(background_colors, axis=0)
+            
+        except Exception as e:
+            logger.error(f"❌ Color analysis failed: {str(e)}")
+            raise RuntimeError(f"Failed to analyze image colors: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)